packages/core/src/PartialExecutionUtils/cleanRunData.ts

```diff
@@ -23,5 +23,14 @@ export function cleanRunData(
 		}
 	}
 
+	// Remove run data for all nodes that are not part of the subgraph
+	for (const nodeName of Object.keys(newRunData)) {
+		const node = graph.getNode(nodeName);
```

Comment:
would it make sense to define and use a `graph.hasNode`, since we don't really use the value of `node` here.

Reply:
Good idea. Done 👍🏾